anyway/current_user_functions.py

```diff
@@ -15,3 +16,9 @@ def get_current_user_email():
     if cur_user is not None:
         return cur_user.email
     return None
+
+
+def get_current_user() -> UserOAuth:
+    cur_id = current_user.get_id()
+    cur_user = db.session.query(UserOAuth).filter(UserOAuth.id == cur_id).first()
```

Comment:
Will work when no user is logged in?

Reply:
No, the caller should check that the user is logged in as this function is returning user specific data

Reply:
So I think return type `Optional[UserOAuth]` will hint users of the method that they need to check for null.

Reply:
What i thought when i wrote this function is that nobody will call it if the user is not logged-in as the function name will indicate that a logged in user is a prerequisite.

Do you think i should add a check for this case?

Reply:
I am not familiar with the usage, but future user may be by one that that is not familiar with this, or in a place in the code were it is not clear whether there is any user logged in currently.  Changing the returned type of `get_current _user()` to `Optional` will hint a future user that it may return None. It is a small cost, and I think it is sufficient.